### PR TITLE
More SPG tweaks

### PR DESCRIPTION
**Pull request by mattandrews**
This pairs with https://github.com/biglotteryfund/grants-service/pull/20.

I noticed that when you apply a sort to your results, it effectively means any future text search will be sorted by whichever sort parameter you applied (eg. not the `textScore`). This change means whenever the text query search is modified, it resets/ignores the previous sort value (eg. so the API knows to order by score).

Also in here: a pairing change to show the total awarded (adds a slightly delay onto the results but not really noticeable, and worth it IMO):

image

There was also a change from last week which sets the `isCalculating` toggle as soon as input is made which means the UI doesn't flicker so much:

### Before:
before

### After:
after

Minor downside is it may look like the search has started as soon as you press a key (whereas in fact the search is still debounced and doesn't start until 500ms after typing):

search

Finally we don't bother outputting default (eg. empty) filters for URLs / AJAX calls, eg:

### Before:

`?amount=&year=&programme=&country=&localAuthority=&constituency=&q=cats`

### After:

`?q=cats`

### CODE REVIEW

File: assets/sass/components/_past-grants.scss [hunk 1]
```diff
@@ -60,9 +60,12 @@
         }
 
         &.is-ascending::before,
-        &.is-ascending::after {
+        &.is-descending::before {
```
Comment (mattandrews) [inline assets/sass/components/_past-grants.scss]: I think this was a typo in the original code?

File: assets/sass/components/_past-grants.scss [hunk 2]
```diff
@@ -60,9 +60,12 @@
...
             position: relative;
             top: 2px; // Optically align
+            margin-right: 4px;
+            display: inline-block;
```
Comment (mattandrews) [inline assets/sass/components/_past-grants.scss]: Added this to fix this:

image

File: controllers/past-grants/views/grant-results.njk [hunk 3]
```diff
@@ -2,7 +2,7 @@
 {% from "components/split-nav/macro.njk" import splitNav with context %}
 
 {% macro grantResultItem(grant) %}
-    <div class="u-margin-bottom">
+    <div class="u-margin-bottom" data-score="{{ grant._textScore }}">
```
Comment (mattandrews) [inline controllers/past-grants/views/grant-results.njk]: (Useful for debugging)

File: assets/js/vue-apps/grant-filters.js [hunk 4]
```diff
@@ -82,9 +91,14 @@ function init() {
 
             // Convert filters into URL-friendly state
             filtersToString: function() {
-                return Object.keys(this.filters)
+                let filterClone = Object.assign({}, this.filters);
```
Comment (davidrapson) [inline assets/js/vue-apps/grant-filters.js]: Maybe use `lodash/cloneDeep` or `lodash/clone` here?

Comment (davidrapson) [reply assets/js/vue-apps/grant-filters.js]: Reading this back this sentence is confusing. I mean something like this:

```js
const filterClone = cloneDeep(this.filters);
const cleanFilters = this.ignoreSort && filterClone.sort ?  omit(filterClone, 'sort') : filterClone;
```

Comment (mattandrews) [reply assets/js/vue-apps/grant-filters.js]: Good shout, give it a look now.

**Pull request merged**